frontend/lib/util/filtering.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021-2023 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/util/filtering.h"
+#include <regex>
+
+namespace chpl {
+
+std::string removeSphinxMarkup(const std::string& msg) {
+  // TODO: Support explicit title and reference targets like in reST direct hyperlinks (and having only target
+  //       show up in sanitized message).
+  static const auto reStr = R"#(\B\:(?:mod|proc|iter|data|const|var|param|type|class|record|attr|enum)\:`(?:([$\w\$\.]+)|(?:~([$\w\$]+\.?)+)|(?:!([$\w\$\.]+)))`\B)#";
+  std::string filteredMsg = msg;
+  std::smatch match;
+  while(std::regex_search(filteredMsg, match, std::regex(reStr))) {
```

Comment:
I get this code is repeatedly applying the regular expression to filter out the Sphinx markup.

I'm not sure what makes this different than what we were doing with `std::regex_replace` before but I'm guessing there's a good reason, so I'd consider adding a comment explaining what that reason is.

Reply:
It is essentially the same as what was before. But now instead of the replacement always being group 2, it could be group 1, 2, or 3. And there is no good way in the C++ regular expression implementation/library to express that. So it requires doing it manually. I will add a comment to this effect.